src/rantanplan/rhymes.py

```diff
@@ -0,0 +1,261 @@
+# -*- coding: utf-8 -*-
+import re
+import statistics
+import string
+
+from spacy_affixes.utils import strip_accents
+
+ASSONANT_RHYME = "assonant"
+CONSONANT_RHYME = "consonant"
+CONSONANTS = r"bcdfghjklmnñpqrstvwxyz"
+UNSTRESSED_VOWELS = r"aeiou"
+STRESSED_VOWELS = r"áéíóúäëïöü"
+WEAK_VOWELS = r"iuïü"
+STRONG_VOWELS = r"aeoáéó"
+WEAK_VOWELS_RE = re.compile(fr'[{WEAK_VOWELS}]([{STRONG_VOWELS}])',
+                            re.U | re.I)
+VOWELS = fr"{UNSTRESSED_VOWELS}{STRESSED_VOWELS}"
+STRESSED_VOWELS_RE = re.compile(fr'[{STRESSED_VOWELS}]', re.U | re.I)
+CONSONANTS_RE = re.compile(fr'[{CONSONANTS}]+', re.U | re.I)
+INITIAL_CONSONANTS_RE = re.compile(fr'^[{CONSONANTS}]+', re.U | re.I)
+DIPHTHONG_H_RE = re.compile(fr'([{VOWELS}])h([{VOWELS}])', re.U | re.I)
+DIPHTHONG_Y_RE = re.compile(fr'([{VOWELS}])h?y([^{VOWELS}])', re.U | re.I)
+# Stanza structures where each tuple is defined as follows:
+# (
+#     CONSONANT_RHYME | ASSONANT_RHYME,
+#     "structure name",
+#     r".*",  # regular expression to match the rhymed line pattern
+#     lambda lengths: True  # function checking a condition on line lengths
+# )
+# Structures will be checked in order of definition, the first one to match
+# will be chosen.
+STRUCTURES = (
+    (
+        CONSONANT_RHYME,
+        "sonnet",
+        r"(abba|abab|cddc|cdcd){2}((cd|ef){3}|(cde|efg){2})",
+        lambda lengths: all(13 > length > 10 for length in lengths)
+    ), (
+        CONSONANT_RHYME,
+        "couplet",
+        r"aa",
+        lambda _: True
+    ), (
+        ASSONANT_RHYME,
+        "romance",
+        r"((.b)+)|((.a)+)",
+        lambda lengths: statistics.median(lengths) == 8
+    ), (
+        ASSONANT_RHYME,
+        "haiku",
+        r".*",
+        lambda lengths: re.compile(r"(575)+").match("".join(
+            [str(length)for length in lengths]
+        ))
+    ), (
+        ASSONANT_RHYME,
+        "couplet",
+        r"aa",
+        lambda _: True
+    ),
+)
+
+
+def get_stressed_endings(lines):
```

Comment:
Maybe we should merge this with my get-sinaeresis function, or probably just change my code to include this one.